### ISSUE TITLE
Aleph's other key trade dialog disappears

### ISSUE THREAD

**Issue by AmProsius** [AmProsius/gothic-1-community-patch#190]
**Describe the bug**
Aleph's dialog choice about the key trade (after the player grassed him up) disappears after loading a savegame, if the player didn't have enough ore.

**Expected behavior**
Aleph's dialog choice about the key trade (after the player grassed him up) no longer disappears, if the player didn't have enough ore.

**Steps to reproduce the issue**
1. Give Brandick the dirt on Aleph.
2. Offer Aleph to buy the key to the chests in the storage shed without enough ore in the inventory.
3. Save the game.
4. Load the save.
5. The dialog choice disappeared.

**Additional context**
https://github.com/AmProsius/gothic-1-community-patch/blob/839bd35f840802afb4884d69a79a7df3bfa050f9/scriptbase/_work/Data/Scripts/Content/Story/Missions/DIA_VLK_585_Aleph.d#L293-L348

**Comment by AmProsius** [comment]
Part of the solution might be to change
https://github.com/AmProsius/gothic-1-community-patch/blob/839bd35f840802afb4884d69a79a7df3bfa050f9/scriptbase/_work/Data/Scripts/Content/Story/Missions/DIA_VLK_585_Aleph.d#L293-L301

to

```d
instance  VLK_585_Aleph_DIRTY (C_INFO)
{
    npc             = VLK_585_Aleph;
    condition       = VLK_585_Aleph_DIRTY_Condition;
    information     = VLK_585_Aleph_DIRTY_Info;
    important       = 0;
    permanent       = 1;
    description     = "Do you know anything about the key for the chests in the storage shed??";
};
```

and

https://github.com/AmProsius/gothic-1-community-patch/blob/839bd35f840802afb4884d69a79a7df3bfa050f9/scriptbase/_work/Data/Scripts/Content/Story/Missions/DIA_VLK_585_Aleph.d#L303-L309

to

```d
FUNC int  VLK_585_Aleph_DIRTY_Condition()
{
    if ( Npc_KnowsInfo (hero, GRD_271_ULBERT_DRUNK) ) 
    && ( Npc_KnowsInfo (hero, GRD_261_Brandick_ALEPH) )
    && ( !Npc_HasItems(hero, ItKe_OM_03) )
    {
        return 1;
    };
};
```

but we will also have to take into account that if the player refused to give him any ore, the dialog choice should also disappear.

https://github.com/AmProsius/gothic-1-community-patch/blob/839bd35f840802afb4884d69a79a7df3bfa050f9/scriptbase/_work/Data/Scripts/Content/Story/Missions/DIA_VLK_585_Aleph.d#L343-L348

Unfortunately I don't know yet how to check if that particular dialog was already spoken.

**Comment by AmProsius** [comment]
See also #40 .

**Comment by szapp** [comment]
I think #40 is independent, because it prevents this dialog to appear when the player obtained the key through the dialog `VLK_585_Aleph_SCHUPPEN` and later ratted out Aleph.

However, after looking at #189 (specifically in [this comment](https://github.com/AmProsius/gothic-1-community-patch/issues/189#issuecomment-827498255)) it looks like there is a better way to check if the player already obtained the key.

If we follow #189, both #40 and this one here will be obsolete.

**Comment by szapp** [comment]
This fix will be taken care of by #189.
